src/index.js

```diff
@@ -59,45 +59,55 @@ io.on('connection', (socket) => {
             user,
             room,
         }
-        generateMessage = makeMessageGenerator(user)
-        generateLocationMessage = makeMessageGenerator(
-            user,
-            (location) => `${location}`,
-            'url'
-        )
 
         socket.join(room)
+        const roomMessages = await roomMessagesFetcher.call(context.room.id)
+        roomMessages.forEach((message) => {
+            if (message.attr.type_id == 1) {
+                io.to(context.room).emit('message', message.attr)
+            } else {
+                io.to(context.room).emit('sendLocation', message.attr)
+            }
+        })
         socket.emit('message', generateWelcome())
```

Comment:
Well, at first I did think about storing them. But then decided not to because:
1) Not all users see them;
2) They don't carry any "sense" in them. You could say that the join/left messages might tell us about the time the user spent in the room, but I didn't think this was an important piece of information so decided not to keep.

Reply:
1) it may be fixed by adding `recipient_id` attribute and filtering by it 
2) agree
Let's do not worry about that

Reply:
Could you please clarify what exactly do you mean by it?
"it may be fixed by adding recipient_id attribute and filtering by it"

The only messages that kind of has a "specific" recipient it the "Welcome" one because it is shown to the user who joins the room.
Other messages are more like "show to everyone" or "show to everyone but the current user".

Reply:
ok, then this task requires additional table for storing visibility mask. Or it can be implemented with one attribute
better not to worry about it